[CoordinatedGraphics] Simplify tiled backing store handling

carlosgcampos opened the pull request:
#### 7c4b2b7e595981864ce559bb62f3bdad0e010ccf
<pre>
[CoordinatedGraphics] Simplify tiled backing store handling
<a href="https://bugs.webkit.org/show_bug.cgi?id=280606">https://bugs.webkit.org/show_bug.cgi?id=280606</a>

Reviewed by Miguel Gomez.

Remove NicosiaBackingStore and add an update method to
CoordinatedBackingStoreProxy that returns an Update object that can be
passed directly to the compositor.

* Source/WebCore/platform/SourcesNicosia.txt:
* Source/WebCore/platform/TextureMapper.cmake:
* Source/WebCore/platform/graphics/nicosia/NicosiaBackingStore.cpp: Removed.
* Source/WebCore/platform/graphics/nicosia/NicosiaBackingStore.h: Removed.
* Source/WebCore/platform/graphics/nicosia/NicosiaCompositionLayer.h:
* Source/WebCore/platform/graphics/nicosia/NicosiaSceneIntegration.cpp:
(Nicosia::SceneIntegration::UpdateScope::~UpdateScope):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStore.cpp:
(WebCore::CoordinatedBackingStore::createTile):
(WebCore::CoordinatedBackingStore::resize):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStore.h:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxy.cpp:
(WebCore::generateTileID):
(WebCore::CoordinatedBackingStoreProxy::Update::Update):
(WebCore::CoordinatedBackingStoreProxy::Update::appendUpdate):
(WebCore::CoordinatedBackingStoreProxy::create):
(WebCore::CoordinatedBackingStoreProxy::CoordinatedBackingStoreProxy):
(WebCore::CoordinatedBackingStoreProxy::setContentsScale):
(WebCore::CoordinatedBackingStoreProxy::updateIfNeeded):
(WebCore::CoordinatedBackingStoreProxy::computeCoverAndKeepRect const):
(WebCore::CoordinatedBackingStoreProxy::invalidateRegion):
(WebCore::CoordinatedBackingStoreProxy::mapToContents const):
(WebCore::CoordinatedBackingStoreProxy::mapFromContents const):
(WebCore::CoordinatedBackingStoreProxy::tileRectForPosition const):
(WebCore::CoordinatedBackingStoreProxy::tilePositionForPoint const):
(WebCore::innerBottomRight): Deleted.
(WebCore::CoordinatedBackingStoreProxy::setTrajectoryVector): Deleted.
(WebCore::CoordinatedBackingStoreProxy::createTilesIfNeeded): Deleted.
(WebCore::CoordinatedBackingStoreProxy::invalidate): Deleted.
(WebCore::CoordinatedBackingStoreProxy::dirtyTiles): Deleted.
(WebCore::CoordinatedBackingStoreProxy::tileDistance const): Deleted.
(WebCore::CoordinatedBackingStoreProxy::coverageRatio const): Deleted.
(WebCore::CoordinatedBackingStoreProxy::visibleAreaIsCovered const): Deleted.
(WebCore::CoordinatedBackingStoreProxy::createTiles): Deleted.
(WebCore::CoordinatedBackingStoreProxy::adjustForContentsRect const): Deleted.
(WebCore::CoordinatedBackingStoreProxy::resizeEdgeTiles): Deleted.
(WebCore::CoordinatedBackingStoreProxy::setKeepRect): Deleted.
(WebCore::CoordinatedBackingStoreProxy::removeAllNonVisibleTiles): Deleted.
(WebCore::CoordinatedBackingStoreProxy::tileRectForCoordinate const): Deleted.
(WebCore::CoordinatedBackingStoreProxy::tileCoordinateForPoint const): Deleted.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxy.h:
(WebCore::CoordinatedBackingStoreProxy::coverRect const):
(WebCore::CoordinatedBackingStoreProxy::hasPendingTileCreation const):
(WebCore::CoordinatedBackingStoreProxy::Tile::Tile):
(WebCore::CoordinatedBackingStoreProxy::Tile::resize):
(WebCore::CoordinatedBackingStoreProxy::Tile::addDirtyRect):
(WebCore::CoordinatedBackingStoreProxy::client): Deleted.
(WebCore::CoordinatedBackingStoreProxy::contentsScale const): Deleted.
(WebCore::CoordinatedBackingStoreProxy::setCoverRect): Deleted.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxyClient.h: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxyTile.cpp: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxyTile.h: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.cpp:
(WebCore::CoordinatedGraphicsLayer::~CoordinatedGraphicsLayer):
(WebCore::CoordinatedGraphicsLayer::flushCompositingStateForThisLayerOnly):
(WebCore::CoordinatedGraphicsLayer::updateContentBuffers):
(WebCore::CoordinatedGraphicsLayer::purgeBackingStores):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.h:
* Source/WebKit/Shared/CoordinatedGraphics/CoordinatedGraphicsScene.cpp:
(WebKit::updateBackingStore):
(WebKit::CoordinatedGraphicsScene::removeLayer):
(WebKit::CoordinatedGraphicsScene::updateSceneState):
(WebKit::CoordinatedGraphicsScene::purgeGLResources):
(WebKit::removeLayer): Deleted.
* Source/WebKit/Shared/CoordinatedGraphics/CoordinatedGraphicsScene.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CompositingCoordinator.cpp:
(WebKit::CompositingCoordinator::flushPendingLayerChanges):

Canonical link: <a href="https://commits.webkit.org/285011@main">https://commits.webkit.org/285011@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5bd92076212d0fc8603495feba7856803569994a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/71221 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/50634 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/23994 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/75329 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/22426 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/73336 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/58434 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/22245 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/56306 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/14756 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/74287 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/45997 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/61371 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/36731 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/42670 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/18833 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/20767 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/64561 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/19196 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/77051 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/15455 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/18385 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/64014 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/15497 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/61406 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/63993 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/12135 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/5761 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10921 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/46434 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/1213 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/47505 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/48788 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/47247 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

magomez commented on Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.cpp:
```diff
@@ -1128,129 +1123,71 @@ std::pair<bool, bool> CoordinatedGraphicsLayer::finalizeCompositingStateFlush()
-    }
+    ASSERT(m_coordinator && m_coordinator->isFlushingLayerChanges());
+
+    auto shouldCreateAndDestroyTiles = m_pendingVisibleRectAdjustment;
```
why create this just to store the value of m_pendingVisibleRectAdjustment for the next call? Why not just passing m_pendingVisibleRectAdjustment to the call and then reset it to false, so we don't need another var?

carlosgcampos replied:
This is me being paranoid because in current code it's changed in that order, but it's not really needed.

carlosgcampos committed:
[CoordinatedGraphics] Simplify tiled backing store handling
https://bugs.webkit.org/show_bug.cgi?id=280606

Reviewed by Miguel Gomez.

Remove NicosiaBackingStore and add an update method to
CoordinatedBackingStoreProxy that returns an Update object that can be
passed directly to the compositor.

* Source/WebCore/platform/SourcesNicosia.txt:
* Source/WebCore/platform/TextureMapper.cmake:
* Source/WebCore/platform/graphics/nicosia/NicosiaBackingStore.cpp: Removed.
* Source/WebCore/platform/graphics/nicosia/NicosiaBackingStore.h: Removed.
* Source/WebCore/platform/graphics/nicosia/NicosiaCompositionLayer.h:
* Source/WebCore/platform/graphics/nicosia/NicosiaSceneIntegration.cpp:
(Nicosia::SceneIntegration::UpdateScope::~UpdateScope):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStore.cpp:
(WebCore::CoordinatedBackingStore::createTile):
(WebCore::CoordinatedBackingStore::resize):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStore.h:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxy.cpp:
(WebCore::generateTileID):
(WebCore::CoordinatedBackingStoreProxy::Update::Update):
(WebCore::CoordinatedBackingStoreProxy::Update::appendUpdate):
(WebCore::CoordinatedBackingStoreProxy::create):
(WebCore::CoordinatedBackingStoreProxy::CoordinatedBackingStoreProxy):
(WebCore::CoordinatedBackingStoreProxy::setContentsScale):
(WebCore::CoordinatedBackingStoreProxy::updateIfNeeded):
(WebCore::CoordinatedBackingStoreProxy::computeCoverAndKeepRect const):
(WebCore::CoordinatedBackingStoreProxy::invalidateRegion):
(WebCore::CoordinatedBackingStoreProxy::mapToContents const):
(WebCore::CoordinatedBackingStoreProxy::mapFromContents const):
(WebCore::CoordinatedBackingStoreProxy::tileRectForPosition const):
(WebCore::CoordinatedBackingStoreProxy::tilePositionForPoint const):
(WebCore::innerBottomRight): Deleted.
(WebCore::CoordinatedBackingStoreProxy::setTrajectoryVector): Deleted.
(WebCore::CoordinatedBackingStoreProxy::createTilesIfNeeded): Deleted.
(WebCore::CoordinatedBackingStoreProxy::invalidate): Deleted.
(WebCore::CoordinatedBackingStoreProxy::dirtyTiles): Deleted.
(WebCore::CoordinatedBackingStoreProxy::tileDistance const): Deleted.
(WebCore::CoordinatedBackingStoreProxy::coverageRatio const): Deleted.
(WebCore::CoordinatedBackingStoreProxy::visibleAreaIsCovered const): Deleted.
(WebCore::CoordinatedBackingStoreProxy::createTiles): Deleted.
(WebCore::CoordinatedBackingStoreProxy::adjustForContentsRect const): Deleted.
(WebCore::CoordinatedBackingStoreProxy::resizeEdgeTiles): Deleted.
(WebCore::CoordinatedBackingStoreProxy::setKeepRect): Deleted.
(WebCore::CoordinatedBackingStoreProxy::removeAllNonVisibleTiles): Deleted.
(WebCore::CoordinatedBackingStoreProxy::tileRectForCoordinate const): Deleted.
(WebCore::CoordinatedBackingStoreProxy::tileCoordinateForPoint const): Deleted.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxy.h:
(WebCore::CoordinatedBackingStoreProxy::coverRect const):
(WebCore::CoordinatedBackingStoreProxy::hasPendingTileCreation const):
(WebCore::CoordinatedBackingStoreProxy::Tile::Tile):
(WebCore::CoordinatedBackingStoreProxy::Tile::resize):
(WebCore::CoordinatedBackingStoreProxy::Tile::addDirtyRect):
(WebCore::CoordinatedBackingStoreProxy::client): Deleted.
(WebCore::CoordinatedBackingStoreProxy::contentsScale const): Deleted.
(WebCore::CoordinatedBackingStoreProxy::setCoverRect): Deleted.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxyClient.h: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxyTile.cpp: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxyTile.h: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.cpp:
(WebCore::CoordinatedGraphicsLayer::~CoordinatedGraphicsLayer):
(WebCore::CoordinatedGraphicsLayer::flushCompositingStateForThisLayerOnly):
(WebCore::CoordinatedGraphicsLayer::updateContentBuffers):
(WebCore::CoordinatedGraphicsLayer::purgeBackingStores):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.h:
* Source/WebKit/Shared/CoordinatedGraphics/CoordinatedGraphicsScene.cpp:
(WebKit::updateBackingStore):
(WebKit::CoordinatedGraphicsScene::removeLayer):
(WebKit::CoordinatedGraphicsScene::updateSceneState):
(WebKit::CoordinatedGraphicsScene::purgeGLResources):
(WebKit::removeLayer): Deleted.
* Source/WebKit/Shared/CoordinatedGraphics/CoordinatedGraphicsScene.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CompositingCoordinator.cpp:
(WebKit::CompositingCoordinator::flushPendingLayerChanges):

Canonical link: https://commits.webkit.org/285011@main
